packages/ui-extensions/src/extension.ts

```diff
@@ -37,10 +37,10 @@ export interface RenderExtensionWithRemoteRoot<
     any
   > = RemoteComponentType<any, any, any>,
 > {
-  (
-    root: RemoteRoot<AllowedComponents, AllowedComponents>,
-    api: Api,
-  ): void | Promise<void>;
+  (root: RemoteRoot<AllowedComponents, AllowedComponents>, api: Api):
+    | void
+    | Promise<() => void>
```

Comment:
Added this return type so that `return new Promise<() => void>` in render.tsx does not cause any type complaints cc @vividviolet

Reply:
Don't we need `void | Promise<void> | Promise<() => void>`? The vanilla version of this don't have a clean up callback

Reply:
Yeah sorry this diff is confusing, the Promise<void> is right under on line 43